rocketpy/simulation/df2flight.py

```diff
@@ -0,0 +1,208 @@
+        property from a data frame. This is made via class because it's the only
+        way to use the __set_name__ method. This method is used to get the name
+        of the property in the class.
+        """
+
+        func = None
+
+        def __init__(self, func):
+            self.attrname = None
+            self.func = func
+
+        def __set_name__(self, owner, name):
+            self.attrname = name
+
+        @wraps(func)
+        def __get__(self, instance, owner=None):
+            """Get method that retrieves the value from the data frame and
+            returns a Function object.
+            """
+            try:
+                name = instance.columns_map[col_name]
+            except KeyError as e:
+                raise KeyError(
+                    f"Cannot get the '{col_name}' variable from the data frame."
+                    + " Please check your columns_map."
+                ) from e
+            x_values = instance.df[name].values
+            source = np.column_stack((instance.time, x_values))
+            return Function(
+                source, "Time (s)", output_name, interpolation, extrapolation
+            )
+
+    return Decorator
+
+
+@dataclass
+class DataFrameToFlight:
+    """A class to create a rocketpy.Flight object from a pandas data frame. Each
+    property of a Flight object will be created using values from the data frame
+
+    There are different ways of calculating a property from the data frame.
+        1. The property is already in data frame and you want to use it as is
+        2. The property is not in the data frame but you can calculate it from
+            other properties in the data frame.
+
+    Initially, this class is designed to work with the option '1.'
+
+    Parameters
+    ----------
+    name : str
+        The name of the flight. This is useful to identify the flight in the
+        simulation. Example: "Rocket Calisto, Spaceport America, 2019"
+    time_col : str
+        The name of the column containing the time values. This is case
+        sensitive. This column is crucial because it is used to create the
+        other rocketpy.Function objects.
+    df : pd.DataFrame
+        The data frame containing the flight data. The data frame must contain
+        the time column.
+    columns_map : dict, optional
+        A dictionary containing the mapping between the column names in the
+        data frame and the names to be used in the Flight object. This is useful
+        when the column names in the data frame are not the same as the ones
+        used in the Flight object. The default is None.
+    interpolation : str, optional
+        The interpolation method to be used in the rocketpy.Function objects.
+        The default is "linear".
+    extrapolation : str, optional
+        The extrapolation method to be used in the rocketpy.Function objects.
+        The default is "zero".
+    """
+
+    name: str
+    time_col: str
+    df: pd.DataFrame
```

Comment:
Pandas is quite a heavy library (> 10 MB while RocketPy is < 300 kB). So, adding it as import (which will significantly slow down importing this file) just to define the type of a data class attribute seems a bit exaggerated to me.

In fact, rumor says that even if you don't import pandas, this line still runs fine since python completely ignores type annotations, as long as what is written is correct syntax wise (see https://death.andgravity.com/dataclasses).

So, I suggest two options:

1. Don't import pandas and keep `df: pd.DataFrame`. Works for python, may cause problems with linters.
2. Don't import pandas and change `df: pd.DataFrame` to `'df: pd.DataFrame'`. Works for python, has less chance of causing problems with linters.

```suggestion
    df: 'pd.DataFrame'
```

Reply:
pandas is no longer necessary